src/Form/Form.js

```diff
@@ -347,12 +381,17 @@ class Form {
             if (foundInputs.length < MAX_INPUTS_PER_FORM) {
                 foundInputs.forEach(input => this.addInput(input))
             } else {
-                if (shouldLog()) {
-                    console.log('The form has too many inputs, bailing.')
-                }
+                // This is rather extreme, but better safe than sorry
+                this.device.scanner.stopScanner('The form has too many inputs, bailing.')
+                return
             }
         }
         this.initialScanComplete = true
+
+        // Observe only if the container isn't the body, to avoid performance overloads
+        if (this.form !== document.body) {
+            this.mutObs.observe(this.form, this.mutObsConfig)
+        }
```

Comment:
Another important part here. The problem happened specifically when autofill got tricked into thinking that the whole `body` was a form (for example, when there is an input field at the top of the page and a bunch of fields down below). This ensures that when that happens we don't append the observer at all. Below there are a couple other measures to ensure we decrease the chance of identifying the `body` as the form.